### DISCUSSION TITLE
useFieldArray URL string become array of object why?

### DISCUSSION THREAD

**Discussion by georgehwc** [react-hook-form/react-hook-form#9253] [Q&A] [answered]
I am trying to use useFieldArray but something weird is coming out and I have no idea, please help.

```  js
const { register, handleSubmit, formState, reset, watch, control, setValue, getValues, trigger, formState: { errors } } = useForm({
    mode: "all",
    defaultValues: activeObject,
  });
  console.log(activeObject);

  const {
    fields: photoFields,
    append: photoAppend,
    remove: photoRemove,
    move: photoMove,
  } = useFieldArray({ control, name: "photo" });
  console.log("photoFields", photoFields);
```
but the console log output is like this, whats wrong here?
![image](https://user-images.githubusercontent.com/46784498/196972659-60fc4dd6-d100-4506-9a61-13674340f1c3.png)

I also can see in `photoFields` the last object is `id`  which is not in the original string. I have no idea where it comes from.
"https://firebasestorage.googleapis.com/v0/b/pet-project-hk.appspot.com/o/ProductImages%2FqvxVeW9nKWzmPvKsJL86%2F10a.jpg2021-12-23?alt=media&token=d587c94b-2b14-4039-b850-443144d446a7"

`
178
: 
"4"
179
: 
"6"
180
: 
"a"
181
: 
"7"
id
: 
"d11ed101-28ed-4331-9906-156e52f94861"
`

**Comment by Moshyfawn** [comment, marked as answer]
Check out this thread regarding the generated ID
https://github.com/react-hook-form/react-hook-form/discussions/8935#discussioncomment-3491228

**Reply by georgehwc** [reply]
Thanks for the thread, so what I understand now, I have to modify my database, because RHF doesnt support array of string, is this correct? or is there any other way to do it?

**Reply by leapful** [reply]
@georgehwc : Could you please provide a codesandbox url to re-produce the issue ?

**Reply by georgehwc** [reply]
@leapful 
https://codesandbox.io/s/bold-star-zh471s?file=/src/App.js
I would like `photoFields` is same as activeObject.photo.
Please check in console, Thanks.

**Reply by georgehwc** [reply]
update: I re-organize my db to be array of object instead of array of string, not good at all but fixed, thanks.

**Reply by Moshyfawn** [reply]
> Thanks for the thread, so what I understand now, I have to modify my database, because RHF doesnt support array of string, is this correct? or is there any other way to do it?

That's correct. `useFieldArray` does not support flat arrays due to the current API design. From RHF docs on [`useFieldArray` options](https://react-hook-form.com/api/usefieldarray#rules):
> Does not support flat field array. 

You can always "shape" your data `onSubmit` to fit your API / DB needs:
```js
const onSubmit = data => {
  const flatArray = data.fields.map(field => field.value)
}
```

**Reply by georgehwc** [reply]
https://codesandbox.io/s/react-beautiful-dnd-react-hook-form-yup-react-dropzone-forked-cskv73?file=/src/App.tsx

actually I am doing something like this but with default value. I had some issue on `draggableId` with my string so I come here, anyway Thanks @Moshyfawn .
